app/src/main/java/comcmput301f16t01/github/carrier/Requests/ElasticRequestController.java

```diff
@@ -128,9 +129,111 @@ protected RequestList doInBackground(String... search_parameters) {
         }
     } // SearchByKeywordTask
 
+    /*
+     * Query:
+     * { "query" : {
+     *     "filtered" : {
+     *       "query" : {
+     *         "bool": {
+     *           "should": [
+     *             { "match": { "status": 1 }},
+     *             { "match": { "status": 2 }}
+     *           ]
+     *         }
+     *       },
+     *       "filter" : {
+     *         "geo_distance" : {
+     *           "distance" : "50km",
+     *           "location" : [-113, 52]
+     *         }
+     *       }
+     *     }
+     *   },
+     *   "sort": [ {
+     *     "_geo_distance": {
+     *       "location": [-113.4909, 53.5444],
+     *       "order": "asc",
+     *       "unit": "km",
+     *       "distance_type": "plane"
+     *     }
+     *   } ]
+     * }
+     */
+
+    /**
+     * Searches requests by a geo-location.
+     */
+    public static class SearchByLocationTask extends AsyncTask<Location, Void, RequestList>{
+
+        /**
+         * Distance represents how far our search query will reach.
```

Comment:
Not my javadoc. But I ask around and find out and add it if you want.
You may have reviewed it after a pull request was merged so you're looking at a non-final product now.